include/drivers/at_cmd.rst

```diff
@@ -0,0 +1,51 @@
+﻿.. _at_cmd_readme:
+
+AT command interface
+####################
+
+The socket interface has a global limitation of 8 socket instances that can be
+open at the same time.
+So if an application needs multiple socket protocols like TCP,
+UDP, TLS, AT, and so on, the available 8 socket instances become a limiting factor.
+The AT command interface driver is designed to let multiple clients, i.e.,
+threads, software modules/components to share a single AT socket instance.
+
+The AT command driver always parses the return code from the modem.
+Non-notification data, if any, such as OK, ERROR, and +CMS/+CME are removed from the string returned to the user.
+The return codes are returned as error codes in the return code of the write function
+and also through the state parameter that can be supplied.
+The state parameter must be used to differentiate between +CMS and +CME errors
+as the error codes are overlapping.
+Any subsequent writes from other threads are queued until all the data (return
+code + any payload) from the previous write is returned to the caller.
+This is to make sure that the correct thread gets the correct data and the return
+code as there is no way to separate two sessions apart.
+The kernel thread priority scheme determines the thread to get write access if multiple threads are queued.
+
+There are two schemes by which data returned immediately from the modem (for instance, the modem response for an AT+CNUM command) is delivered to the user.
+The user can submit either a reference to a string buffer or a reference to a handler function in the write function.
```

Comment:
Doesn't the user need to call different functions, depending on how they want the data returned? This sentence makes it sound like the distinction is made based on what the user submits to one function.

Reply:
Modified as:
 The user can call the write function by submitting either of the following in the write function:...